### PR TITLE
Fix _CRS checking non-compound CRS for sub-CRS objects

### LINKED ISSUE #822: Pyproj 3+ logs PROJ_ERROR at debug level
#### Code Sample, a copy-pastable example if possible

A "Minimal, Complete and Verifiable Example" will make it much easier for maintainers to help you:
http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports

```python
import logging
logging.basicConfig(level=logging.DEBUG)

from pyproj import Proj
p = Proj("+proj=stere +a=6370000 +b=6370000 +lat_0=90 +lat_ts=60 +lon_0=10 +units=m +no_defs")
p.crs.is_geographic
# DEBUG:pyproj:PROJ_ERROR: proj_crs_get_sub_crs: Object is not a CompoundCRS
```

Note: The message doesn't show up with the creation of CRS, only `is_geographic`. There are likely other cases that cause the message but this was the simplest I could find.

#### Problem description

I use pyproj in a lot of applications that include logging. I often write a log file with DEBUG level logging. Starting with pyproj 3, I'm seeing a lot of the following error messages that are logged as DEBUG messages and are filling up my log file. I'm hoping for more information on what these mean, if they are meant to be logged, and what the right course of action might be to reducing them.

```
DEBUG:pyproj:PROJ_ERROR: proj_crs_get_sub_crs: Object is not a CompoundCRS
```

#### Expected Output

No log message or at least not a DEBUG message with a PROJ_ERROR.

#### Environment Information

 - Output from: `pyproj -v`

```
pyproj info:
    pyproj: 3.0.1
      PROJ: 8.0.0
  data dir: /home/davidh/miniconda3/envs/polar2grid_py38/share/proj
user_data_dir: /home/davidh/.local/share/proj

System:
    python: 3.8.8 | packaged by conda-forge | (default, Feb 20 2021, 16:22:27)  [GCC 9.3.0]
executable: /home/davidh/miniconda3/envs/polar2grid_py38/bin/python
   machine: Linux-5.11.0-7612-generic-x86_64-with-glibc2.10

Python deps:
       pip: 20.2.3
setuptools: 49.6.0.post20200814
    Cython: None
```

#### Installation method
 - conda-forge

#### Conda environment information (if you installed with conda):

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list proj

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

```
</details>

### PR DESCRIPTION
**Pull request by djhoese**
See #822 for more details. Bottom line `_CRS.sub_crs_list` was checking for sub-CRSs (how do you pluralize an acronym?) even on non-compound CRS objects. This PR adds an `is_compound` property to `_CRS` so that it can then check it in `sub_crs_list` and avoid the additional logic.

* Where should I put tests (I didn't see any specifically for is_bound)?
* Do I add information to history.rst under 3.1.0?

 - [x] Closes #822 
 - [x] Tests added
 - [x] Fully documented, including `history.rst` for all changes and `api/*.rst` for new API

### CODE REVIEW

**Comment by snowman2** [comment]
Thanks! Mind exposing `is_compound` on the main CRS class as well?
https://github.com/pyproj4/pyproj/blob/9669f623596ef19ba7cf2754639814ccc81cc4a9/pyproj/crs/crs.py#L1366-L1374

**Comment by snowman2** [comment]
> Where should I put tests (I didn't see any specifically for is_bound)?

I would put it in `/tests/crs/test_crs.py`:
```python 
assert CRS("EPSG:4326+5773").is_compound
```

**Comment by snowman2** [comment]
> Do I add information to history.rst under 3.1.0?

Yeah, that would be the spot for it.

File: pyproj/_crs.pyx [hunk 1]
```diff
@@ -2925,6 +2929,16 @@ cdef class _CRS(Base):
         """
         return self._type == PJ_TYPE_BOUND_CRS
 
+    @property
+    def is_compound(self):
+        """
```
Comment (snowman2) [inline pyproj/_crs.pyx:2934]: Mind adding this?
```
        .. versionadded:: 3.1
```

**Comment by djhoese** [comment]
Will do on all those changes. Testing now. For API docs, ever thought about using https://pypi.org/project/sphinxcontrib-apidoc/ to generate them automatically?

**Comment by snowman2** [comment]
> For API docs, ever thought about using https://pypi.org/project/sphinxcontrib-apidoc/ to generate them automatically?

Looks worthwhile to look into. Do you use it for satpy?

**Comment by djhoese** [comment]
Yep: https://github.com/pytroll/satpy/blob/master/doc/source/conf.py#L81-L92. Basically end up gitignore'ing the entire `api/` directory. This also assumes you don't customize the order of the modules in the `.rst` files or any other complex changes..

**Comment by snowman2** [comment]
This looks good. Ready to merge?

**Comment by djhoese** [comment]
Yep, go for it.

**Comment by snowman2** [comment]
Thanks @djhoese 👍

**Pull request merged**